### PR TITLE
Implement hello world using attribute

### PR DESCRIPTION
**Pull request by yjaeseok**
First, I changed hello.nidl readonly attribute to operation.
because operation is simple more than attribute.

And then I copied type transit header at bacardi project.
But, I wonder we should keep our copyright, or not.

Last I implemented template like bacardi.
I refer to https://github.com/lunchclass/bacardi/blob/master/template/interface_cpp.njk

Please review @romandev , @nadongguri

### CODE REVIEW

File: generator/templates/interface_cpp.tmpl [hunk 1]
```diff
@@ -1,12 +1,16 @@
 #include "nidl_<%= data.name.toLowerCase() %>.h"
 
-void NIDL_<%= data.name %>::Init(Napi::Env env, Napi::Object exports) {
-  Napi::Function js_constructor =
-      DefineClass(env, "<%= data.name %>",
-          {
-            InstanceAccessor("world", &NIDL_<%= data.name %>::GetterWorld, nullptr),
-          });
+#include "../../generator/js_type_traits.h"
```
Comment (yjaeseok) [inline generator/templates/interface_cpp.tmpl]: If we cannot use relative path, I would like to change it

**Comment by yjaeseok** [comment]
Result nidl_hello.h

``` c++
#ifndef EXAMPLES_HELLO_NIDL_H_
#define EXAMPLES_HELLO_NIDL_H_

#include <memory>
#include <napi.h>

#include "../hello.h"

#include "../../generator/js_type_traits.h"
#include "../../generator/native_type_traits.h"

class NIDL_Hello : public Napi::ObjectWrap<NIDL_Hello> {
 public:
  static void Init(Napi::Env env, Napi::Object exports);

  explicit NIDL_Hello(const Napi::CallbackInfo& info);



    Napi::Value GetterWorld(const Napi::CallbackInfo& info);
    void SetterWorld(const Napi::CallbackInfo& info, const Napi::Value& value);


 private:
  std::unique_ptr<Hello> impl_;
};

#endif  // EXAMPLES_HELLO_NIDL_H_

```

**Comment by yjaeseok** [comment]
Result nidl_hello.cc

``` c++
#include "nidl_hello.h"

#include "../../generator/js_type_traits.h"
#include "../../generator/native_type_traits.h"

void NIDL_Hello::Init(Napi::Env env, Napi::Object exports) {
  Napi::Function js_constructor = DefineClass(env, "Hello", {


      InstanceAccessor("world", &NIDL_Hello::GetterWorld, &NIDL_Hello::SetterWorld),


  });
  exports.Set("Hello", js_constructor);
}

NIDL_Hello::NIDL_Hello(const Napi::CallbackInfo& info)
    : Napi::ObjectWrap<NIDL_Hello>(info) {
  impl_.reset(new Hello());
}



  Napi::Value NIDL_Hello::GetterWorld(const Napi::CallbackInfo& info) {
    auto return_value = impl_->GetWorld();
    return JSTypeTraits(info.Env(), return_value);
  }
  void NIDL_Hello::SetterWorld(const Napi::CallbackInfo& info, const Napi::Value& value) {
    impl_->SetWorld(NativeTypeTraits<IDLString>::NativeValue(info.Env(), info[0]));
  }
```

**Comment by yjaeseok** [comment]
hello.js

``` javascript
const buildType = process.config.target_defaults.default_configuration;
const nidl = require(`./build/${buildType}/hello.node`);

const hello = new nidl.Hello();
hello.world = "Hello, world!";
console.log(hello.world); // 'Hello, world!'
```

File: generator/templates/interface_header.tmpl [hunk 2]
```diff
@@ -6,14 +6,23 @@
 
 #include "../<%= data.name.toLowerCase() %>.h"
 
+#include "../../generator/js_type_traits.h"
```
Comment (romandev) [inline generator/templates/interface_header.tmpl]: These should be "non-relative" path. To resolve this issue, we can add the path to `include` object in index.js. Please see this line: https://github.com/romandev/node-addon-api/blob/master/test/binding.gyp#L26

File: generator/idl_base.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (romandev) [inline generator/idl_base.h]: Could you please move these C++ files to other directory? I'd like to put them with `.js` files separately.

**Pull request merged**